web-client/src/presenter/actions/CaseDeadline/updateCaseDeadlineAction.ts

```diff
@@ -22,6 +23,16 @@ export const updateCaseDeadlineAction = async ({
       caseDeadline,
     });
 
+  const existingCaseDeadlines = get(state.caseDeadlines);
+
+  const updatedCaseDeadlines = existingCaseDeadlines.map(deadline =>
+    deadline.caseDeadlineId === updateCaseDeadlineResult.caseDeadlineId
+      ? updateCaseDeadlineResult
+      : deadline,
+  );
+
+  store.set(state.caseDeadlines, updatedCaseDeadlines);
```

Comment:
This assumes that `caseDeadlines` is in state. Since the only place we use this action is on the Tracked Items tab of the Case Detail page, and since we have loaded the `caseDeadlines` when we come to this page, this assumption should be ok.